src/requests/UsersApi.ts

```diff
@@ -0,0 +1,192 @@
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Users.Models.User>(paramRequest);
+
+        return queryResult;
+    }
+
+    /**
+     * Get Groups a user is in by their User ID.
+     * @param userId The id of the user to get information about.
+     * @returns The information about the user. If the user is not found then it will return undefined.
+     */
+    async getUserGroups({ userId }: VRCAPI.Users.Requests.getUserGroupsByUserIdOptions): Promise<VRCAPI.Groups.Models.Group[]> {
+
+        const parameters: URLSearchParams = new URLSearchParams();
+
+        if (typeof userId !== 'string') {
+            throw new Error('No user id was provided!');
+        }
+        if (userId.length < 1) {
+            throw new Error('Empty User id or no User id were provided!');
+        }
+
+        parameters.append('userId', userId);
+
+        const paramRequest: VRCAPI.Generics.executeRequestType = {
+            currentRequest: ApiPaths.users.getUserGroups,
+            pathFormated: ApiPaths.users.getUserGroups.path.replace('{userId}', userId),
+            queryOptions: parameters,
+        };
+
+        this.checkValidData(paramRequest);
+        const queryResult = await this.executeRequest<VRCAPI.Groups.Models.Group[]>(paramRequest);
+
+        return queryResult;
+    }
+
+}
+
+    /**
+     * This function is used to get the rank tag of a user. Returns the highest rank tag of the user. Else returns undefined.
+     * @param user The User or CurrentUser object to get the rank tag from.
+     * @returns The rank tag of the user. If the user has no rank tag, returns undefined.
+     */
+    export function getVRCRankTag(user: VRCAPI.Users.Models.User | VRCAPI.Users.Models.CurrentUser): VRCAPI.Users.Models.VRCRanks | undefined {
+        // the highest vrcrank we can find in the user's tag is the rank of the user we should return
+        const tags: VRCAPI.Generics.AllTags[] = user.tags;
+        if (user.tags.includes(VRCRanks.Veteran)) {
+            return VRCRanks.Veteran;
+        } else if (tags.includes(VRCRanks.Trusted)) {
+            return VRCRanks.Trusted;
+        } else if (tags.includes(VRCRanks.Known)) {
+            return VRCRanks.Known;
+        } else if (tags.includes(VRCRanks.User)) {
+            return VRCRanks.User;
+        } else if (tags.includes(VRCRanks.New)) {
+            return VRCRanks.New;
+        } else if (!tags.includes(VRCRanks.Veteran) &&
+            !tags.includes(VRCRanks.Trusted) &&
+            !tags.includes(VRCRanks.Known) &&
+            !tags.includes(VRCRanks.User) &&
+            !tags.includes(VRCRanks.New) &&
+            !tags.includes(VRCRanks.Nuisance)) {
+
+            return VRCRanks.Visitor;
+        } else if (tags.includes(VRCRanks.Nuisance)) {
+            return VRCRanks.Nuisance;
+        } else {
+            return undefined;
+        }
```

Comment:
???? Really?

Reply:
Think really hard on what you wrote here.

Reply:
that was an old code from my last comit lol it just got moved along I will redo this lol

Reply:
> ???? Really?

thinking back on it, there is a very good reason I did it the way I did, I could explain to you by VC, but technically, vrchat users have multiple tags, I need to find the first one the user has. I could maybe optimize this little code but it does what I want it to do in the right order.

Reply:
I ended up fixing completely this function as well as the two bellow it 😉